plugins/modules/aci_qos_dscp_class.py

```diff
@@ -0,0 +1,348 @@
+from ansible_collections.cisco.aci.plugins.module_utils.aci import (
+    ACIModule,
+    aci_argument_spec,
+    aci_annotation_spec,
+    aci_owner_spec,
+    aci_contract_dscp_spec,
+)
+from ansible_collections.cisco.aci.plugins.module_utils.constants import MATCH_TARGET_COS_MAPPING
+
+
+def main():
+    new_dscp_spec = dict((k, aci_contract_dscp_spec()[k]) for k in aci_contract_dscp_spec() if k != "aliases")
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        qos_custom_policy=dict(type="str", aliases=["qos_custom_policy_name"]),
+        priority=dict(
+            type="str",
+            choices=[
+                "level1",
+                "level2",
+                "level3",
+                "level4",
+                "level5",
+                "level6",
+                "unspecified",
+            ],
+            aliases=["prio"],
+        ),
+        dscp_from=new_dscp_spec,
+        dscp_to=new_dscp_spec,
+        dscp_target=aci_contract_dscp_spec(),
+        target_cos=dict(type="str", choices=list(MATCH_TARGET_COS_MAPPING.keys())),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "qos_custom_policy", "dscp_from", "dscp_to"]],
+            ["state", "present", ["tenant", "qos_custom_policy", "dscp_from", "dscp_to"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    qos_custom_policy = module.params.get("qos_custom_policy")
+    priority = module.params.get("priority")
+    dscp_from = module.params.get("dscp_from")
+    dscp_to = module.params.get("dscp_to")
+    dscp_target = module.params.get("dscp_target")
+    target_cos = MATCH_TARGET_COS_MAPPING.get(module.params.get("target_cos"))
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="qosCustomPol",
+            aci_rn="qoscustom-{0}".format(qos_custom_policy),
+            module_object=qos_custom_policy,
+            target_filter={"name": qos_custom_policy},
+        ),
+        subclass_2=dict(
+            aci_class="qosDscpClass",
+            aci_rn="dcsp-{0}-{1}".format(dscp_from, dscp_to),
+            module_object=qos_custom_policy,
+            target_filter={"from": dscp_from, "to": dscp_to},
+        ),
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        aci.payload(
+            aci_class="qosDscpClass",
+            class_config={
```

Comment:
class_config = dict(
prio=priority,
from_=dscp_from,
to=dscp_to,
target=dscp_target,
targetCos=target_cos,
)